src/store/src/Indexer.php

```diff
@@ -26,57 +26,37 @@
 class Indexer implements IndexerInterface
 {
     /**
-     * @var array<string|null>
-     */
-    private array $sources = [];
-
-    /**
-     * @param string|array<string>|null $source       Source identifier(s) for data loading (file paths, URLs, etc.)
-     * @param FilterInterface[]         $filters      Filters to apply after loading documents to remove unwanted content
-     * @param TransformerInterface[]    $transformers Transformers to mutate documents after filtering (chunking, cleaning, etc.)
+     * @param FilterInterface[]      $filters      Filters to apply after loading documents to remove unwanted content
+     * @param TransformerInterface[] $transformers Transformers to mutate documents after filtering (chunking, cleaning, etc.)
      */
     public function __construct(
-        private LoaderInterface $loader,
+        private Loader $loader,
         private VectorizerInterface $vectorizer,
         private StoreInterface $store,
-        string|array|null $source = null,
         private array $filters = [],
         private array $transformers = [],
         private LoggerInterface $logger = new NullLogger(),
     ) {
-        $this->sources = null === $source ? [] : (array) $source;
-    }
-
-    public function withSource(string|array $source): self
-    {
-        return new self($this->loader, $this->vectorizer, $this->store, $source, $this->filters, $this->transformers, $this->logger);
     }
 
-    public function index(array $options = []): void
+    public function index(iterable|SourceInterface $sources, array $options = []): void
```

Comment:
so now, we must pass something when indexing? We loose the ability to run :
```bash
bin/console ai.store.index indexer_name
```
Or I missed something?

Reply:
yup, that's open and needs to be tackled in the bundle :+1: will follow